src/app/auth/login.ts

```diff
@@ -113,6 +128,171 @@ export const loginWithPhoneCookie = async ({
+    // is there still txns left over on the device account?
+    const deviceWallets = await WalletsRepository().listByAccountId(account.id)
+    if (deviceWallets instanceof Error) return deviceWallets
+    const ledger = LedgerService()
+    let deviceAccountHasBalance = false
+    for (const wallet of deviceWallets) {
+      const balance = await ledger.getWalletBalance(wallet.id)
+      if (balance instanceof Error) return balance
+      if (balance > 0) {
+        deviceAccountHasBalance = true
+      }
+    }
+    if (deviceAccountHasBalance) {
+      /// Scenario 1.5 - has txns on device account but phone account exists
+      return new PhoneAccountAlreadyExistsNeedToSweepFunds(
+        `Error phone account already exists. You need to manually sweep funds
+        to your phone account. Here are the steps to do this:
+        1. login with phone account and create an invoice (save this invoice)
+        2. logout of the phone account
+        3. login with device account (click "Explore Wallet Instead" button)
+        4  Pay the invoice with max wallet amount.
+        5. Logout of the device account
+        6. Log into the phone account and check for the funds`,
+      )
+    } else {
+      /// Scenario 1 - no txns on device account but phone account exists
+      return new PhoneAccountAlreadyExists(
+        "Phone Account already exists. Please logout and log back in with your phone account.",
+      )
+    }
+  }
+
+  // Scenario 2 - Happy Path
+  // a. update kratos
+  const kratosAccount = await AccountsRepository().findByUserId(account.kratosUserId)
+  if (kratosAccount instanceof Error) return kratosAccount
+  const updatedKratosAccount = await AuthWithDeviceAccountService().upgradeToPhoneSchema({
+    kratosUserId: account.kratosUserId,
+    phone,
+  })
+  if (updatedKratosAccount instanceof Error) return new Error("updatedKratosAccount")
+  // b. update user
+  const userRemovedDevice = await UsersRepository().unsetDeviceIdForUser(
+    account.kratosUserId,
+  )
+  if (userRemovedDevice instanceof Error) return userRemovedDevice
+  const userAddedPhone = await UsersRepository().update({
+    id: account.kratosUserId,
+    phone,
+  })
+  if (userAddedPhone instanceof Error) return userAddedPhone
+  // c. update account
+  kratosAccount.level = AccountLevel.One
+  const accountLevelOne = await AccountsRepository().update(kratosAccount)
+  if (accountLevelOne instanceof Error) return accountLevelOne
+
+  // returning a new session token
+  const authPhone = AuthWithPhonePasswordlessService()
+  const kratosResult = await authPhone.loginToken({ phone })
+  if (kratosResult instanceof Error) return kratosResult
+  return kratosResult.sessionToken
+}
+
+export const loginWithDevice = async ({
+  jwt,
+  ip,
+}: {
+  jwt: string
+  ip: IpAddress
+}): Promise<SessionToken | ApplicationError> => {
+  {
+    const limitOk = await checkFailedLoginAttemptPerIpLimits(ip)
+    if (limitOk instanceof Error) return limitOk
+  }
+
```

Comment:
we are no longer using `createDeviceAccountGlobal` in the code right now.

Reply:
removed here https://github.com/GaloyMoney/galoy/pull/2071/commits/abc51de5b4c012442eeb7f9bd1800fbdb2c7efbf